File: array/1769_minimum_number_of_operations_to_move_all_balls_to_each_box/1769_minimum_number_of_operations_to_move_all_balls_to_each_box.go
package main

import "fmt"

func minOperations(boxes string) []int {
	res := []int{}
	n := len(boxes)
	var total, now int
	for i := 0; i < n; i++ {
		total += now
		res = append(res, total)
		if boxes[i] == '1' {
			now++
		}
	}
	total, now = 0, 0
	for i := n - 1; i >= 0; i-- {
		total += now
		res[i] += total
		if boxes[i] == '1' {
			now++
		}
	}
	return res

}

func main() {
	boxes := "110"
	res := minOperations(boxes)
	fmt.Println(res)
}

File: array/1769_minimum_number_of_operations_to_move_all_balls_to_each_box/1769_minimum_number_of_operations_to_move_all_balls_to_each_box.py
from typing import List


class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        ans = []
        left = right = total = 0
        n = len(boxes)
        for i in range(n):
            if boxes[i] == '1':
                right += 1
                total += i
        print(right, total)
        for i in range(n):
            ans.append(total)
            if boxes[i] == '1':
                left += 1
                right -= 1
            total += left - right
        return ans


boxes = "110"
res = Solution().minOperations(boxes)
print(res)